### PR TITLE
Cut 1.8 for the changes to the withGauge function

### PR DESCRIPTION
**Pull request by tomcarste**
Major version bump for the breaking changes in `withGauge`.

### CODE REVIEW

**Comment by tomcarste** [comment]
Decided to go for a stop gap type class similar to HasSqlPool. Mostly because getting the Maybe right in a lens produced an abhorrent instance for XRayVaultData* and not having the Maybe requires changes to the underlying library. In the end our goal is to have our middleware take care of handling non-present VaultData and we can make this a regular lens has field type class.

```vaultDataL f x = fmap (const x) $ f (Just x)```

I'm sure there has to be something better than this.

**Comment by tomcarste** [comment]
Also @pbrisbin any instances that we should supply by default?

File: library/Freckle/App/Database.hs [hunk 1]
```diff
@@ -54,7 +55,7 @@ import System.Process.Typed (proc, readProcessStdout_)
 import UnliftIO.Concurrent (threadDelay)
 import UnliftIO.Exception (displayException)
 import UnliftIO.IORef
-import Yesod.Core (MonadHandler, MonadUnliftIO(withRunInIO), waiRequest)
+import Yesod.Core (MonadUnliftIO (withRunInIO))
```
Comment (pbrisbin) [inline library/Freckle/App/Database.hs:58]: :art: 
```suggestion
import Yesod.Core (MonadUnliftIO(withRunInIO))
```

File: library/Freckle/App/Test.hs [hunk 2]
```diff
@@ -103,6 +103,8 @@ instance HasLogger app => Example (AppExample app a) where
     params
     ($ ())
 
+instance HasVaultData app => HasVaultData (AppExample app a) where
+  getVaultData = asks getVaultData
```
Comment (pbrisbin) [inline library/Freckle/App/Test.hs:107]: :art: 
```suggestion
  getVaultData = asks getVaultData
  
```

**Review by pbrisbin — COMMENTED** [review]
I think, to use this from a `HandlerFor app`, you'll want:

```hs
instance HasVaultData (HandlerData child site)
  getVaultData = vaultDataFromRequest . reqWaiRequest . handlerRequest
```

File: library/Freckle/App/Database.hs [hunk 3]
```diff
@@ -74,15 +82,15 @@ makePostgresPool = do
...
+     , HasVaultData app
      , MonadUnliftIO m
      , MonadReader app m
      )
   => SqlPersistT m a
   -> m a
 runDB action = do
   pool <- asks getSqlPool
-  mVaultData <- vaultDataFromRequest <$> waiRequest
+  mVaultData <- asks getVaultData-- vaultDataFromRequest <$> waiRequest
```
Comment (pbrisbin) [inline library/Freckle/App/Database.hs:93]:
```suggestion
  mVaultData <- asks getVaultData
```

**Review by pbrisbin — APPROVED** [review]

**Pull request merged**